app/display.h

```diff
@@ -36,57 +37,76 @@
     prints(0, 30, str)
 
 #define display_cpu_clk(freq) \
-    printf(1, 10, "%i MHz", freq)
+    printf(1, 10, "%iMHz", freq)
 
 #define display_cpu_addr_mode(str) \
-    prints(1, 20, str)
+    prints(5, 76, str)
 
 #define display_l1_cache_size(size) \
-    printf(2, 10, "%kB", (uintptr_t)(size))
+    printf(2, 9, "%6kB", (uintptr_t)(size));
 
 #define display_l2_cache_size(size) \
-    printf(3, 10, "%kB", (uintptr_t)(size))
+    printf(3, 9, "%6kB", (uintptr_t)(size));
 
 #define display_l3_cache_size(size) \
-    printf(4, 10, "%kB", (uintptr_t)(size))
+    printf(4, 9, "%6kB", (uintptr_t)(size));
+
+#define display_memory_size(size) \
+    printf(5, 9, "%6kB", (uintptr_t)(size));
 
 #define display_l1_cache_speed(size) \
-    printf(2, 19, "%kB/s", (uintptr_t)(size))
+    printf(2, 18, "%S6kB/s", (uintptr_t)(size))
```

Comment:
The parameter name should be "speed" not "size" (also in following definitions).